File: seed/php-model/pagination/src/Users/ListUsersTopLevelCursorPaginationResponse.php
<?php

namespace Seed\Users;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

class ListUsersTopLevelCursorPaginationResponse extends JsonSerializableType
{
    /**
     * @var ?string $nextCursor
     */
    #[JsonProperty('next_cursor')]
    public ?string $nextCursor;

    /**
     * @var array<User> $data
     */
    #[JsonProperty('data'), ArrayType([User::class])]
    public array $data;

    /**
     * @param array{
     *   data: array<User>,
     *   nextCursor?: ?string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->nextCursor = $values['nextCursor'] ?? null;
        $this->data = $values['data'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/ruby-sdk-v2/pagination/lib/seed/users/types/list_users_top_level_cursor_pagination_response.rb
# frozen_string_literal: true

module Seed
  module Users
    module Types
      class ListUsersTopLevelCursorPaginationResponse < Internal::Types::Model
        field :next_cursor, -> { String }, optional: true, nullable: false
        field :data, -> { Internal::Types::Array[Seed::Users::Types::User] }, optional: false, nullable: false
      end
    end
  end
end
